src/routes/threlte/stores.ts

```diff
@@ -0,0 +1,125 @@
+export const realization = writable<Realization>(emptyRealization())
+export const realizationIsStale = writable(false)
+
+export const hiddenSketches = writable<string[]>([])
+export const sketchBeingEdited = writable("")
+export const sketchTool = writable("")
+
+// could be looking for 'face' or 'plane' or other things
+export const selectingFor = writable<EntityType[]>([])
+export const selectionMax = writable(1000)
+export const selectionMin = writable(0)
+
+export const currentlyMousedOver = writable<SnapEntity[]>([])
+export const currentlySelected = writable<Entity[]>([])
+export const snapPoints = writable<PointLikeById[]>([])
+export const previewGeometry = writable<PreviewGeometry[]>([])
+
+export const messageHistory = writable<MessageHistory[]>([])
+
+project.subscribe(store => log("[project]", store))
+workbenchIndex.subscribe(store => log("[workbenchIndex]", store))
+workbench.subscribe(store => log("[workbench]", store))
+workbenchIsStale.subscribe(store => log("[workbenchIsStale]", store))
+featureIndex.subscribe(store => log("[featureIndex]", store))
+extrusionFeatures.subscribe(store => log("[extrusionFeatures]", store))
+realization.subscribe(store => log("[realization]", store))
+realizationIsStale.subscribe(store => log("[realizationIsStale]", store))
+sketchBeingEdited.subscribe(store => log("[sketchBeingEdited]", store))
+messageHistory.subscribe(store => log("[messageHistory]", store))
+
+if (_isDevelopment()) {
+  currentlySelected.subscribe((store) => {
+    log("[currentlySelected]", store)
+    const allValid = store.every(entity => isEntity(entity))
+    const error = "[stores.ts] [currentlySelected] has invalid entities"
+    if (!allValid) {
+      console.error(error, store)
+      // throw new Error(error)
+    }
+
+    const types = ["circle", "arc", "face", "line", "plane", "point", "point3D", "meshFace"] as EntityType[]
+    types.forEach(type => {
+      const [isType, entity] = latestIsEntity(store, type)
+      if (isType) log(`[currentlySelected] entity is ${type === "arc" ? "an" : "a"} ${type}:`, entity)
+    })
+  })
+}
+
+function latestIsEntity(store: Entity[], type: EntityType) {
+  if (store.length === 0) return [false, null]
+  const entity = store[store.length - 1]
+  switch (type) {
+    case "circle":
+      return [isCircleEntity(entity), entity]
+    case "arc":
+      return [isArcEntity(entity), entity]
+    case "face":
+      return [isFaceEntity(entity), entity]
+    case "line":
+      return [isLineEntity(entity), entity]
+    case "plane":
+      return [isPlaneEntity(entity), entity]
+    case "point":
+      return [isPointEntity(entity), entity]
+    case "point3D":
+      return [isPoint3DEntity(entity), entity]
+    case "meshFace":
+      return [isMeshFaceEntity(entity), entity]
+    default:
+      break
+  }
+  log("[latestIsEntity] has incorrect switch statement implemented")
+  return [false, null]
+}
+
+function emptyWorkBench(): WorkBench {
```

Comment:
this pattern seems like a bummer to me. No need to change it now or anything, but I wonder what we could do to not have to write these kinds of shadow objects in the future.

Reply:
yeah it's a bit ugly, but typescript complains, rightfully so when the values aren't initialised on store creation. we could turn them into pojo when creating the `writable` stores? I actually put them down here as functions so they would be out of sight! could be moved off to a detestable.ts file too, but I agree there's no need to do anything with it today. a cleaner solution will happen naturally at a later date when working on this file I expect.